internal/aws/cwlogs/cwlog_client.go

```diff
@@ -175,19 +199,23 @@ func (client *Client) CreateStream(logGroup, streamName *string) error {
 	return nil
 }
 
-func newCollectorUserAgentHandler(buildInfo component.BuildInfo, logGroupName string, componentName string) request.NamedHandler {
+func newCollectorUserAgentHandler(buildInfo component.BuildInfo, logGroupName string, componentName string, userAgentFlag *UserAgentFlag) request.NamedHandler {
+	extraStr := ""
+
+	switch {
+	case containerInsightsRegexPattern.MatchString(logGroupName):
+		extraStr = "ContainerInsights"
+	case userAgentFlag.isAppSignals:
+		extraStr = "AppSignals"
+	}
```

Comment:
With the suggestions I have made previously I don't think it has to be one or the other.